internal/provider/errors.go

```diff
@@ -0,0 +1,26 @@
+package provider
+
+import "strings"
+
+type errSet struct {
```

Comment:
maybe using this package would be easier https://github.com/hashicorp/go-multierror